service-config/src/main/java/com/palantir/config/service/ServiceDiscoveryConfiguration.java

```diff
@@ -136,6 +136,20 @@
         return getServiceConfig(serviceName).readTimeout();
     }
 
+    /**
+     * Checks if a service is enabled.
+     * <p>
+     * A service is enabled iff the configuration for it exists and its list of URIs is non-empty.
+     *
+     * @param serviceName the name of the service
+     * @return whether or not the service is enabled
+     */
+    public final boolean isServiceEnabled(String serviceName) {
```

Comment:
should this and other attributes here be marked as `@Value.Derived`? http://immutables.github.io/immutable.html#derived-attributes